action.yml

```diff
@@ -0,0 +1,83 @@
+name: Setup Python and Poetry
+description: Setup Python and Poetry. Cribbed from snok/install-poetry.
```

Comment:
Do we need to carry any licensing info across, and/or perhaps link to it in a hypothetical README?

Reply:
"Cribbed" is maybe the wrong word. I looked to see how that action worked, but in the end decided I wanted different behaviour and wrote most of the yaml from scratch. snok's action was very useful for understanding what is/was going on with poetry though.

I'll acknowledge this and mark this as MIT licensed to match snok/install-poetry. I know we normally use Apache 2, but given that this is CI config rather than a core software component I think that's fine.